### PR TITLE
Add IJava recipe

### PR DESCRIPTION
**Pull request by mkitti**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mkitti** [comment]
@conda-forge/help-java This should be a noarch package.

File: recipes/ijava/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+requirements:
+  build:
+    # If your project compiles code (such as a C extension) then add the required compilers as separate entries here.
+    # Compilers are named 'c', 'cxx' and 'fortran'.
+    - openjdk <18
+    - python
+    - unzip  # [not win]
+    - jupyter
+  host:
```
Comment (mkitti) [inline recipes/ijava/meta.yaml:44]:
```suggestion
    - jupyter
  host:
    - python
```

File: recipes/ijava/bld.bat [hunk 2]
```diff
@@ -0,0 +1,6 @@
+echo on
+call .\gradlew zipKernel
+tar -xvf "build\distributions\ijava-%PKG_VERSION%.zip"
+dir /s
```
Comment (mkitti) [inline recipes/ijava/bld.bat:4]:
```suggestion
dir /s
echo on
```

File: recipes/ijava/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    - python
```
Comment (mkitti) [inline recipes/ijava/meta.yaml:45]:
```suggestion
    - python
    - jupyter
```

**Comment by mkitti** [comment]
@conda-forge/help-java I have tests passing on all platforms now. Please take a look. It would be great to have a Java kernel for Jupyter on conda-forge!

**Comment by mkitti** [comment]
@conda-forge/core Could we please add this Jupyter kernel for Java?

File: recipes/ijava/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
```
Comment (ocefpaf) [inline recipes/ijava/meta.yaml:1]: Ping

Comment (mkitti) [reply recipes/ijava/meta.yaml]: Pong. Thank you for the reminder. Comments removed in https://github.com/conda-forge/staged-recipes/commit/60228f19bb03adc3849ffac8abba04768a5feea9

**Pull request merged**